src/main/java/com/cst438/service/RegistrarServiceProxy.java

```diff
@@ -0,0 +1,311 @@
+        }
+        sectionRepository.save(section);
+    }
+    private void updateSection(String s) throws Exception {
+        SectionDTO sectionDTO = fromJsonString(s, SectionDTO.class);
+        Section section = sectionRepository.findById(sectionDTO.secNo()).orElse(null);
+
+        if (s==null) {
+            throw new Exception("section not found "+sectionDTO.secNo());
+        }
+        section.setSecId(sectionDTO.secId());
+        section.setBuilding(sectionDTO.building());
+        section.setRoom(sectionDTO.room());
+        section.setTimes(sectionDTO.times());
+
+        User instructor = null;
+        if (sectionDTO.instructorEmail()==null || sectionDTO.instructorEmail().equals("")) {
+            section.setInstructor_email("");
+        } else {
+            instructor = userRepository.findByEmail(sectionDTO.instructorEmail());
+            if (instructor == null || !instructor.getType().equals("INSTRUCTOR")) {
+                throw new Exception("email not found or not an instructor " + sectionDTO.instructorEmail());
+            }
+            section.setInstructor_email(sectionDTO.instructorEmail());
+        }
+        sectionRepository.save(section);
+    }
+    private void deleteSection(String s) throws Exception {
+        try {
+            Integer id = Integer.parseInt(s);
+            sectionRepository.deleteById(id);
+        } catch (NumberFormatException e) {
+            throw new Exception("Section contains assignments or enrollments: " + s);
+        }
+
+    }
+    private void addUser(String s) throws Exception {
+        UserDTO dto = fromJsonString(s, UserDTO.class);
+        User u = new User();
+        u.setEmail(dto.name());
+        u.setName(dto.email());
+
+        u.setType(dto.type());
+        if (!dto.type().equals("STUDENT") &&
+            !dto.type().equals("INSTRUCTOR") &&
+            !dto.type().equals("ADMIN")) {
+            // invalid type
+            throw new Exception("invalid user type");
+        }
+        userRepository.save(u);
+    }
+    private void updateUser(String s) throws Exception {
+        UserDTO dto = fromJsonString(s, UserDTO.class);
+        User u = userRepository.findById(dto.id()).orElse(null);
+        if (u==null) {
+            throw new Exception("user id not found");
+        }
+        u.setName(dto.name());
+        u.setEmail(dto.email());
+        u.setType(dto.type());
+        if (!dto.type().equals("STUDENT") &&
+            !dto.type().equals("INSTRUCTOR") &&
+            !dto.type().equals("ADMIN")) {
+            // invalid type
+            throw new Exception("invalid user type");
+        }
+        userRepository.save(u);
+    }
+    private void deleteUser(String s) throws Exception {
+            try {
+                Integer id = Integer.parseInt(s);
+                userRepository.deleteById(id);
+            } catch (NumberFormatException e) {
+                throw new Exception("User not found: " + s);
+            }
```

Comment:
This is kind of nitpicky, but the try catch is catching an exception from a string that cannot be parsed to Int. The exception message would be more accurate if it said something like ```Malformed integer, cannot parse to int```. You can also check that the id exists in the DB and throw an exception if it does not. See ```deleteCourse``` for an example. I just pushed error handling for it.